libs/components/forms/src/lib/modules/file-attachment/file-drop/file-drop.component.ts

```diff
@@ -411,17 +475,18 @@ export class SkyFileDropComponent implements OnDestroy {
     this.#emitFileChangeEvent(totalFiles, rejectedFileArray, validFileArray);
   }
 
-  #loadFile(
+  readonly #fileReaderSvc = inject(SkyFileReaderService);
+
+  async #loadFile(
     fileDrop: SkyFileDropComponent,
     file: SkyFileItem,
     validFileArray: SkyFileItem[],
     rejectedFileArray: SkyFileItem[],
     totalFiles: number,
-  ): void {
-    const reader = new FileReader();
+  ): Promise<void> {
+    try {
+      file.url = await this.#fileReaderSvc.readAsDataURL(file.file);
```

Comment:
using the new SkyFileReaderService so that I can mock it out in the unit test and call `SetValue` which calls `WriteValue`